common/browser.go

```diff
@@ -208,130 +220,133 @@ func (b *Browser) initEvents() error {
 	return nil
 }
 
+// onAttachedToTarget is called when a new page is attached to the browser.
```

Comment:
It's a little difficult to follow what's going on here. Do you think you could revert the refactoring (splitting the large method into smaller ones), so that we can see the actual changes to the `onAttachedToTarget` logic, and then have the refactoring in another commit?

Reply:
Sorry for the mess. You're right. I'll split the commits and let you know.

Reply:
Nicely done. very easy to follow now, thanks!